docs/topics/power-assert.md

```diff
@@ -0,0 +1,380 @@
+
+```text
+Assertion failed
+assert(isValidName && isValidAge)
+       |              |
+       |              false
+       true
+```
+
+Inline the variable into the `assert()` function:
+
+```kotlin
+class ComplexExampleTest {
+
+    data class Person(val name: String, val age: Int)
+
+    @Test
+    fun testComplexAssertion() {
+        val person = Person("Alice", 30)
+        assert(person.name.startsWith("A") && person.name.length > 3 && person.age > 20 && person.age < 29)
+    }
+}
+```
+
+After execution, you get more explicit information about what went wrong:
+
+```text
+Assertion failed
+assert(person.name.startsWith("A") && person.name.length > 3 && person.age > 20 && person.age < 29)
+       |      |    |                  |      |    |      |      |      |   |       |      |   |
+       |      |    |                  |      |    |      |      |      |   |       |      |   false
+       |      |    |                  |      |    |      |      |      |   |       |      30
+       |      |    |                  |      |    |      |      |      |   |       Person(name=Alice, age=30)
+       |      |    |                  |      |    |      |      |      |   true
+       |      |    |                  |      |    |      |      |      30
+       |      |    |                  |      |    |      |      Person(name=Alice, age=30)
+       |      |    |                  |      |    |      true
+       |      |    |                  |      |    5
+       |      |    |                  |      Alice
+       |      |    |                  Person(name=Alice, age=30)
+       |      |    true
+       |      Alice
+       Person(name=Alice, age=30)
+```
+
+### Beyond assert function
+
+The Power-assert plugin can transform various functions beyond `assert`.
+Functions like `require()`, `check()`, `assertTrue()`, `assertEqual()` and others can also be transformed,
+if they have a form that allows taking a `String` or `() -> String` value as the last parameter.
+
+Before adding a new function in a test, specify the function in the `powerAssert {}` block of your build script file.
+For example, the `require()` function:
+
+```kotlin
+// build.gradle.kts
+@OptIn(ExperimentalKotlinGradlePluginApi::class)
+powerAssert {
+    functions = listOf("kotlin.assert", "kotlin.require")
+}
+```
+
+After adding the function, you can use it in your tests:
+
+```kotlin
+class RequireExampleTest {
+
+    @Test
+    fun testRequireFunction() {
+        val value = ""
+        require(value.isNotEmpty()) { "Value should not be empty" }
+    }
+}
+```
+
+The output for this example uses the Power-assert plugin to provide detailed information about the failed test:
+
+```text
+Value should not be empty
+require(value.isNotEmpty()) { "Value should not be empty" }
+        |     |
+        |     false
+        
+```
+
+The message shows the intermediate values that lead to the failure, making it easier to debug.
+
+### Function call tracing
```

Comment:
I don't think this example is equivalent to the one from kotlin-power-assert. What might have been missing is I excluded the explicit configuring of the `dbg` function from the kotlin-power-assert example for brevity, which may have been a mistake. Let's consider the other example again for a moment.

```kotlin
fun <T> dbg(value: T): T = value

fun <T> dbg(value: T, msg: String): T {
    println(msg)
    return value
}

fun main() {
    println(dbg(1 + 2 + 3))
}
```

with output:

```text
dbg(1 + 2 + 3)
      |   |
      |   6
      3
6
```

This example basically shows that a function doesn't always need to throw an exception to be useful. Instead, we call the `dbg(T)` function, but power-assert rewrites the call to the `dbg(T, String)` function and includes the call diagram. This allows us to "log" the call parameters with `println`.

So if you have the `dbg` function configured with power-assert, it's sort of like `println` that you can wrap stuff with and get it automatically logged to the console with contextual information. This is more of a main code example, rather than a test code example. Hopefully that explains better what this example is attempting to demonstrate.

Reply:
Maybe I'm doing something wrong, but it doesn't work for me.
I specify the `dbg` function in the `powerAssert {}` block but `println()` shows me the result without any intermediate output.

Reply:
Hmm... interesting. To avoid delaying documentation, I would say exclude this example for now and we can add it later with some additional documentation on what the compiler plugin expects for function signatures. It's kind of an advanced use case that I doubt many people will be interested in right away anyways.

Reply:
Ok, I'll comment this one